buildmaster/ATLAS_2JET_7TEV_R06/filter.py

```diff
@@ -0,0 +1,329 @@
+# Filter for ATLAS_2JET_7TEV_R06
+"""
+Created on Mar  2023
+
+@author: Mark N. Costantini
+"""
+
+import yaml
+import numpy as np
+import pandas as pd
+from scipy.linalg import block_diag, cholesky
+
+
+from validphys.covmats import dataset_inputs_covmat_from_systematics
+from validphys.loader import Loader
+import warnings
+warnings.filterwarnings('ignore')
+
+
+
+def filter_ATLAS_2JET_7TEV_R06(scenario=0,kin_data=True):
+    """
+    Parameters
+    ----------
+
+    scenario : 0, 1, 2
+            0 = nominal, 1 = stronger, 2 = weaker
+            default is 0
+    
+    kin_data : bool
+            whether to save kin and data
+    
+    """
+
+    with open('metadata.yaml', 'r') as file:
+        metadata = yaml.safe_load(file)
+
+    version = metadata['hepdata']['version']
+    tables  = metadata['hepdata']['tables']
+
+    data_central = []
+    kin = []
+    dfs = []
+    for table in tables:
+        hepdata_tables = f"rawdata/HEPData-ins1268975-v{version}-Table_{table}.yaml"
+        
+        # uncertainties dataframe
+        df = fill_df(hepdata_tables,scenario=scenario)
+        dfs.append(df)
+
+        with open(hepdata_tables, 'r') as file:
+            input = yaml.safe_load(file)
+        
+        ystar = range_str_to_floats(input['dependent_variables'][0]['qualifiers'][8]['value'])
+        sqrts = float(input['dependent_variables'][0]['qualifiers'][7]['value'])
+
+        # measurements of several dijet mass values
+        values = input['dependent_variables'][0]['values']
+
+        # loop over different diijet mass bins
+        for i,value in enumerate(values):
```

Comment:
A very slight improvement would be something like
```python
m12_values = input['independent_variables'][0]['values']
for value, m12 in zip(values, m12_values)
```

so you don't need the i index. But is is really a minor thing.